akka-stream/src/main/scala/akka/stream/impl/fusing/GraphInterpreter.scala

```diff
@@ -675,7 +694,7 @@ private[stream] final class GraphInterpreter(
     val currentState = portStates(connection)
     if (Debug) println(s"$Name   complete($connection) [$currentState]")
     portStates(connection) = currentState | OutClosed
-    if ((currentState & (InClosed | Pushing | Pulling)) == 0) enqueue(connection)
+    if ((currentState & (InClosed | Pushing | Pulling | OutClosed)) == 0) enqueue(connection)
```

Comment:
@drewhk this one and the following one are bugfixes: enqueueing the connection again may lead to queue overflow if the previous completion has not yet been processed